kedro/extras/datasets/api/api_dataset.py

```diff
@@ -74,7 +73,7 @@ def __init__(
         headers: Dict[str, Any] = None,
         auth: Union[Tuple[str], AuthBase] = None,
         timeout: int = 60,
-        json: bool = False,
+        attribute: str = "text",
```

Comment:
I'm not entirely sold on the name of `attribute` or its use. What about just being either `json` or `response` and returning either `json` or the response object directly, having to access the attribute of `response` within the node isn't a big deal IMO.